src/connect.js

```diff
@@ -1,24 +1,26 @@
-import React, { PropTypes, Component } from 'react';
+import React, {Component} from 'react';
 import shallowCompare from 'react-addons-shallow-compare';
 
-import storeShape from './storeShape';
+import storeShape from './storeShape.js';
 
-export default function connect(mapToProps, helperProp = 'helper') {
-  return Composed => {
-    return class Connected extends Component {
-      static contextTypes = {
-        algoliaStore: storeShape.isRequired,
-      };
+const getDisplayName = WrappedComponent =>
+  WrappedComponent.displayName || WrappedComponent.name || 'UnknownComponent';
+
+export default function connect(mapStateToProps) {
+  return WrappedComponent =>
+    class Connect extends Component {
+      static contextTypes = {algoliaStore: storeShape.isRequired};
+      static displayName = `AlgoliaSearchHelperConnect(${getDisplayName(WrappedComponent)})`;
 
       constructor(props, context) {
-        super();
+        super(props, context);
```

Comment:
You don't need to pass props or context to `super`. React automatically sets them on the component instance anyway right after the constructor is called. The only effect calling `super` with props and context has is setting them on the instance earlier, so that you can access them via `this.props` and `this.context` in the constructor. Not really useful since you can already access them via `props` and `context`.